src/classes/OPP_OpportunityNaming.cls

```diff
@@ -0,0 +1,392 @@
+    * @description Calculates opportunity names based on Opportunity Naming Settings using the primary 
+    * Contact Id for Role field for contact fields.
+    * @param oppList The list of Opportunities to rename.
+    * @return list<Opportunity> A new list of renamed opportunities. The new list comes from an Opportunity
+    * query and can be updated in an after opportunity insert trigger context.
+    ********************************************************************************************************/
+    public static list<Opportunity> getOppNamesAfterInsert(list<Opportunity> oppList) {
+        
+        set<string> allFields = new set<string>();
+
+        fillOppNameSettings();
+
+        if (listOppNamingSettings.isEmpty())
+            return new list<Opportunity>();
+
+        for (Opportunity_Naming_Settings__c ons : listOppNamingSettings)
+            allFields.addAll(getFieldNamesFromSpec(ons.Opportunity_Name_Format__c));
+
+        list<Opportunity> oppsForUpdate = database.query(getOppQuery(allFields) + ' WHERE Id IN :oppList');
+        
+        set<id> setConId = new set<id>();
+        for (Opportunity opp : oppsForUpdate) {
+            if (opp.npe01__Contact_Id_for_Role__c != null) {
+                setConId.add(opp.npe01__Contact_Id_for_Role__c);
+            }
+        }
+
+        map<id,Contact> mapCons = new map<id,Contact>();
+        if (!setConId.isEmpty()) {
+            mapCons = new map<id,Contact>((list<Contact>)database.query(getConQuery(allFields) + ' WHERE Id IN :setConId'));
+        }
+
+        for (Opportunity opp : oppsForUpdate) {
+
+            //use the naming setting for this opportunity if it exists, otherwise use the default
+            Opportunity_Naming_Settings__c ons;
+            if (mapOppNameSettings.containsKey(getOppAttribution(opp) + opp.RecordType.Name)) {
+                ons = mapOppNameSettings.get(getOppAttribution(opp) + opp.RecordType.Name);
+            } else if (mapOppNameSettings.containsKey(getOppAttribution(opp))) {
+                ons = mapOppNameSettings.get(getOppAttribution(opp));
+            }
+
+            if (ons!=null && !string.isBlank(ons.Opportunity_Name_Format__c))
+                opp.Name = replaceFieldNamesWithValues(opp, mapCons.get(opp.npe01__Contact_Id_for_Role__c), ons);
+            
+            if (opp.Name.length() > 120)
+                opp.Name = opp.Name.substring(0,120);
+
+        }
+
+        return oppsForUpdate;
+    }
+
+    /*******************************************************************************************************
+    * @description Fills the mapOppNameSettings map with record type and attribution info and all settings.
+    ********************************************************************************************************/
+    public static void fillOppNameSettings() {
+        listOppNamingSettings = Opportunity_Naming_Settings__c.getAll().clone().values();
```

Comment:
Although in the settings page controller, where I'm listing all the settings, I do a deepclone as I want to modify some of the values displayed on the page but not the actual values of the settings.